packages/react/src/users/hooks/useUserCredit.ts

```diff
@@ -0,0 +1,139 @@
+    fetchUserCreditMovementsAction,
+  );
+  const userCredits = getUserCredits(user);
+  const isLoading = useSelector(areUserCreditsLoading);
+  const error = useSelector(getUserCreditsError);
+  const isFetched = useSelector(areUserCreditsFetched);
+  const userCreditMovements = getUserCreditMovements(user);
+  const areUserCreditMovementsLoading = useSelector(
+    areUserCreditMovementsLoadingSelector,
+  );
+  const userCreditMovementsError = useSelector(
+    getUserCreditMovementsErrorSelector,
+  );
+  const areUserCreditMovementsFetched = useSelector(
+    areUserCreditMovementsFetchedSelector,
+  );
+
+  /**
+   * Fetches all user credits.
+   * Will throw an error if the user is not authenticated.
+   *
+   * @param config - Custom configurations to send to the client instance (axios).
+   *
+   * @returns - Promise that will be resolved when the call to the endpoint finishes.
+   */
+  const fetch = useCallback(
+    (config: Config | undefined = fetchConfig) => {
+      if (!isAuthenticated) {
+        return Promise.reject(new Error('User is not authenticated'));
+      }
+
+      return fetchUserCredits(userId as User['id'], config);
+    },
+    [fetchUserCredits, isAuthenticated, userId, fetchConfig],
+  );
+
+  /**
+   * Fetches all user credit movements.
+   * Will throw an error if the user is not authenticated.
+   *
+   * @param config - Custom configurations to send to the client instance (axios).
+   *
+   * @returns - Promise that will be resolved when the call to the endpoint finishes.
+   */
+  const fetchUserCreditMovements = useCallback(
+    (config: Config | undefined = fetchConfig) => {
+      if (!isAuthenticated) {
+        return Promise.reject(new Error('User is not authenticated'));
+      }
+
+      return fetchUserCreditMovementsActionDispatcher(
+        userId as User['id'],
+        queryUserCreditMovements,
+        config,
+      );
+    },
+    [
+      fetchConfig,
+      isAuthenticated,
+      fetchUserCreditMovementsActionDispatcher,
+      userId,
+      queryUserCreditMovements,
+    ],
+  );
+
+  useEffect(() => {
+    const updatedState = store.getState() as StoreState;
+
+    const updatedIsLoading = areUserCreditsLoading(updatedState);
+    const updatedIsFetched = areUserCreditsFetched(updatedState);
+
+    if (
+      enableAutoFetch &&
+      !updatedIsFetched &&
+      !updatedIsLoading &&
+      isAuthenticated
+    ) {
+      fetch(fetchConfig);
+    }
+  }, [enableAutoFetch, fetchConfig, fetch, isAuthenticated, isLoading, store]);
+
+  const data = useMemo(() => {
+    if (!userCredits && !userCreditMovements) {
+      return undefined;
+    }
+
+    return { userCredits, userCreditMovements };
+  }, [userCredits, userCreditMovements]);
+
+  return {
+    data,
+    isLoading,
+    error,
+    isFetched,
+    areUserCreditMovementsLoading,
+    userCreditMovementsError,
+    areUserCreditMovementsFetched,
+    actions: {
+      fetch,
+      fetchUserCreditMovements,
```

Comment:
I think we can name the actions and data related to user credit movements to be more concise like `fetchMovements` and `areMovementsLoading`, etc...

Reply:
Changed it 👍

Reply:
Thank you!